framework/src/utils/MooseEnum.C

```diff
@@ -46,62 +43,56 @@ MooseEnum::withNamesFrom(const MooseEnumBase & other_enum)
 /**
  * Private constuctor for use by libmesh::Parameters
  */
-MooseEnum::MooseEnum() : _current_id(INVALID_ID) {}
+MooseEnum::MooseEnum() : _current("", INVALID_ID) {}
 
-MooseEnum::MooseEnum(const MooseEnumBase & other_enum) : MooseEnumBase(other_enum) {}
+MooseEnum::MooseEnum(const MooseEnumBase & other_enum)
+  : MooseEnumBase(other_enum), _current("", INVALID_ID)
```

Comment:
The reason behind this if the "withNamesFrom" static method that creates a MooseEnum/MultiMooseEnum with the names from another. I am going to try to remove this method all together to see what happens.